webapp/channels/src/components/admin_console/access_control/editors/cel_editor/editor.tsx

```diff
@@ -0,0 +1,363 @@
+import TestResultsModal from '../../modals/policy_test/test_modal';
+import {TestButton, HelpText} from '../shared';
+
+import './editor.scss';
+
+export const POLICY_LANGUAGE = 'expressionLanguage';
+const VALIDATE_POLICY_SYNTAX_COMMAND_ID = 'policyEditorValidateSyntaxCommand';
+
+const MONACO_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions = {
+    extraEditorClassName: 'policyEditor',
+    language: POLICY_LANGUAGE,
+    automaticLayout: true,
+    minimap: {enabled: false},
+    lineNumbers: 'off',
+    scrollBeyondLastLine: false,
+    wordWrap: 'on',
+    renderLineHighlight: 'none',
+    lineNumbersMinChars: 1,
+    occurrencesHighlight: 'off',
+    stickyScroll: {enabled: false},
+    autoClosingBrackets: 'never',
+    autoClosingQuotes: 'never',
+    autoIndent: 'keep',
+    autoSurround: 'never',
+    codeLens: false,
+    folding: false,
+    fontFamily: 'monospace',
+    hideCursorInOverviewRuler: true,
+    fontSize: 12,
+    guides: {indentation: false},
+    links: true,
+    matchBrackets: 'never',
+    multiCursorLimit: 1,
+    overviewRulerBorder: false,
+    quickSuggestions: false,
+    renderControlCharacters: false,
+    scrollbar: {
+        horizontal: 'hidden',
+        useShadows: false,
+    },
+    selectionHighlight: false,
+    showFoldingControls: 'never',
+    suggestOnTriggerCharacters: true,
+    unicodeHighlight: {
+        ambiguousCharacters: false,
+        invisibleCharacters: false,
+    },
+    unusualLineTerminators: 'auto',
+    wordWrapColumn: 400,
+    wrappingIndent: 'none',
+    wrappingStrategy: 'advanced',
+    contextmenu: false,
+};
+
+interface CELEditorProps {
+    value: string;
+    onChange: (value: string) => void;
+    onValidate?: (isValid: boolean) => void;
+    placeholder?: string;
+    className?: string;
+    userAttributes: Array<{
+        attribute: string;
+        values: string[];
+    }>;
+}
+
+// TODO: this is just a sample schema for the editor, we need to get the actual schema from the server
+
+function CELEditor({
+    value,
+    onChange,
+    onValidate,
+    placeholder = 'user.attributes.<attribute> == <value>',
+    className = '',
+    userAttributes,
+}: CELEditorProps): JSX.Element {
+    const [editorState, setEditorState] = useState({
+        expression: value,
+        isValidating: false,
+        isValid: true,
+        cursorPosition: {line: 1, column: 1},
+        validationErrors: [] as string[],
+        statusBarColor: 'var(--button-bg)',
+        showTestResults: false,
+        testResults: null as AccessControlTestResult | null,
+    });
+
+    const schemas = {
+        user: ['attributes'],
+        'user.attributes': userAttributes.map((attr) => attr.attribute),
+    };
+
+    const editorRef = useRef(null);
+    const monacoRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+    const [showHelpModal, setShowHelpModal] = useState(false);
+
+    useEffect(() => {
```

Comment:
(Post-merge) Similar to my comment on the table editor, I think we should change the value prop to `initialValue` to prevent changes made outside of this from accidentally overwriting the user's progress unless there's some reason we know the outside code needs to do that (like clicking a reset button or something).

If we remove that, we may also be able to remove the next `useEffect` that sets the value on the Monaco instance